java/java.source.base/src/org/netbeans/modules/java/source/NoJavacHelper.java

```diff
@@ -18,99 +18,34 @@
-        try {
-            Class.forName("com.sun.tools.javac.comp.Repair");
-            return true;
-        } catch (ClassNotFoundException ex) {
-            //OK
-            return false;
-        }
+    public static boolean hasWorkingJavac() {
+        return hasWorkingJavac;
     }
 
-    @OnStart
-    public static class FixClasses implements Runnable {
-
-        @Override
-        public void run() {
-            if (!hasWorkingJavac()) {
-                String JavaVersion = System.getProperty("java.specification.version"); //NOI18N
-                boolean isJdkVer8OrBelow = true;
-                if (!JavaVersion.startsWith("1.")) {   //NOI18N
-                    isJdkVer8OrBelow = false;
-                }
-                if (isJdkVer8OrBelow) {
-                    {
-                        ClassWriter w = new ClassWriter(0);
-                        w.visit(Opcodes.V1_8, Opcodes.ACC_ABSTRACT | Opcodes.ACC_PUBLIC, "com/sun/tools/javac/code/Scope$WriteableScope", null, "com/sun/tools/javac/code/Scope", null);
-                        byte[] classData = w.toByteArray();
-
-                        defineClass("com.sun.tools.javac.code.Scope$WriteableScope",
-                                    "com.sun.tools.javac.code.Scope",
-                                    classData);
-                    }
-                    {
-                        ClassWriter w = new ClassWriter(0);
-                        w.visit(Opcodes.V1_8, Opcodes.ACC_ABSTRACT | Opcodes.ACC_INTERFACE | Opcodes.ACC_PUBLIC, "javax/lang/model/element/ModuleElement", null, "java/lang/Object", new String[] {"javax/lang/model/element/Element"});
-                        byte[] classData = w.toByteArray();
-
-                        defineClass("javax.lang.model.element.ModuleElement",
-                                    "com.sun.tools.javac.code.Scope",
-                                    classData);
-                    }
-                }
-            }
-        }
-
-        private void defineClass(String fqn, String injectToClass, byte[] classData) {
-            try {
-                Class<?> unsafeClass = Class.forName("sun.misc.Unsafe");
-                Field theUnsafe = unsafeClass.getDeclaredField("theUnsafe"); //NOI18N
-                theUnsafe.setAccessible(true);
-                Object unsafe = theUnsafe.get(null);
-
-                Class targetClass = Class.forName(injectToClass);  //NOI18N
-
-                Method defineClass = unsafeClass.getDeclaredMethod("defineClass", String.class, byte[].class, int.class, int.class, ClassLoader.class, ProtectionDomain.class);  //NOI18N
-                defineClass.invoke(unsafe, fqn, classData, 0, classData.length, targetClass.getClassLoader(), targetClass.getProtectionDomain());  //NOI18N 
-            } catch (Throwable t) {
-                //ignore...
-                Logger.getLogger(NoJavacHelper.class.getName()).log(Level.WARNING, null, t);
-            }
-        }
+    @Deprecated
```

Comment:
I think I'd simply remove this method. This is not an API, we can remove whatever we want. (Maybe just increment the implementation version in `java.source.base/manifest.mf`, so that old and new modules won't mix.)

Reply:
will do.